### PR TITLE
Worker

### PR DESCRIPTION
**Pull request by KaranRam245**
Manager leest data van csv en maakt een taskpool van(totaal 200 tasks). De worker kan een task nemen en een prediction doen hierop. De input csv data is op de drive te vinden. Gedeeld met iedereen

### CODE REVIEW

File: src/aws/nodeworker/nodeworker.py [hunk 1]
```diff
@@ -53,15 +56,28 @@ async def process(self):
...
                     file = self.storage_connector.download_file(
                         file_path=self.current_task.file_path,
                         key=self.current_task.key
                     )
                     log_info("Downloaded file " + self.current_task.file_path + ".")
+                    # file = self.storage_connector.download_file(
+                    #     file_path=self.current_task.file_path,
+                    #     key=self.current_task.key
+                    # )
+                    input_data = self.current_task(0).get_task_data()
```
Comment (demasterr) [inline src/aws/nodeworker/nodeworker.py:70]: `current_task` is currently a  CommandPacket. How does `current_task(0)` work? And then how does `get_task_data()` are you planning to implement this later on?

File: src/aws/nodeworker/nodeworker.py [hunk 2]
```diff
@@ -53,15 +56,28 @@ async def process(self):
...
+                    input_sequences = Tokenize.tokenize_text(
+                        os.path.join("models", "tokenizer_20000.pickle"), input_data)
+                    model = load_model(os.path.join("models", "Senti.h5"))
+                    labels = model.predict(input_sequences)
```
Comment (demasterr) [inline src/aws/nodeworker/nodeworker.py:74]: What is the result of the model? Is the result labels?

File: src/data/Tokenize.py [hunk 3]
```diff
@@ -22,3 +22,10 @@ def train_tokenizer(filePath, maxVocabSize, maxSequenceLength):
     data = sequence.pad_sequences(sequences, maxlen=maxSequenceLength)
 
     return data, targets, tokenizer.word_index
+
+
+def tokenize_text(tokenizer_path, text):
+    with open(tokenizer_path) as handle:
```
Comment (demasterr) [inline src/data/Tokenize.py:28]: You open `tokenizer_path` here but do not use handle. Is this `with` statement even needed then? Doesn't `pickle.load` handle the file opening?

File: src/aws/nodeworker/nodeworker.py (hunk 2 above)
Comment (KaranRam245) [reply src/aws/nodeworker/nodeworker.py:74]: The result is a binary np array 1x6 i.e. [0,1,0,1,1,1]. This maps to the sentiments. Should this value be returned?

File: src/aws/nodeworker/nodeworker.py (hunk 1 above)
Comment (KaranRam245) [reply src/aws/nodeworker/nodeworker.py:70]: Oooh this is my misunderstanding. I missed the part where current_task is a command packet. I thought this was a task object which contains the get_task_data() method. I will adjust for this

File: src/data/Tokenize.py (hunk 3 above)
Comment (KaranRam245) [reply src/data/Tokenize.py:28]: Good question. You have good point. This is stack overflow code so I didnt pay much attention once it worked

File: src/aws/nodeworker/nodeworker.py (hunk 1 above)
Comment (demasterr) [reply src/aws/nodeworker/nodeworker.py:70]: 👍

File: src/aws/nodeworker/nodeworker.py (hunk 2 above)
Comment (demasterr) [reply src/aws/nodeworker/nodeworker.py:74]: We need to know the precision or accuracy or some metric to know how good the model is. This was said in the experiment. We could also choose to remove this from the paper but it makes our story a bit more complete

**Pull request closed without merging**